### PR TITLE
Expose box shape

### PR DESCRIPTION
**Pull request by andens**
This allows creating box shapes that can be used for rigid bodies or triggers.

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Shape.hpp#L100](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d7a45edb72837d029fc7098253bc66ffeffc86a9/src/Engine/Physics/Shape.hpp#L100): Class 'Shape' is unsafe, 'Shape::shape' can leak by wrong usage. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunsafeClassCanLeak)
1. ![INFO][INFO] [EntityEditor.cpp#L446](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d7a45edb72837d029fc7098253bc66ffeffc86a9/src/Editor/GUI/Editors/EntityEditor.cpp#L446): keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Editor/GUI/Editors/BoxShapeEditor.hpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    class BoxShapeEditor : public IShapeEditor {
+        public:
+            /// Constructor
+            BoxShapeEditor() = default;
+
+            /// Box type label.
+            /**
+             * @return "Box" string.
+             */
+            virtual const char* Label() const override { return "Box"; }
```
Comment (Chainsawkitten) [inline src/Editor/GUI/Editors/BoxShapeEditor.hpp]: `virtual` is redundant due to `override`

File: src/Engine/Component/Shape.cpp [hunk 2]
```diff
@@ -8,29 +9,39 @@ namespace Component {
 
         Json::Value concreteShape;
         switch (shape->GetKind()) {
-            case ::Physics::Shape::Kind::Sphere: {
+            case Physics::Shape::Kind::Sphere: {
```
Comment (Chainsawkitten) [inline src/Engine/Component/Shape.cpp]: Why the brackets?

File: src/Editor/GUI/Editors/BoxShapeEditor.hpp (hunk 1 above)
Comment (andens) [reply src/Editor/GUI/Editors/BoxShapeEditor.hpp]: Fixed. Removed the other occurences in the same file and also in the plane- and sphere shape editors.

File: src/Engine/Component/Shape.cpp (hunk 2 above)
Comment (andens) [reply src/Engine/Component/Shape.cpp]: Scoping. I declare local variables that would otherwise leak to subsequent cases.

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![INFO][INFO] [EntityEditor.cpp#L446](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/1a1301c6387f2c218cbdeccc9c21e31b184b26e2/src/Editor/GUI/Editors/EntityEditor.cpp#L446): keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**